source/arduino.h

```
@@ -552,12 +552,109 @@ void tone(int pin, unsigned int frequency, unsigned long duration);
 void noTone(int pin);
 
 //
+// Interrupt functions.
+//
+
+/// Attach a callback routine to a GPIO interrupt.
+/**
+\param[in] pin The number of the board pin for which interrupts are wanted.
+\param[in] fund The function to be called when an interrupt occurs for the specified pin.
+\param[in] mode The type of pin state changes that should cause interrupts.
+*/
+inline void attachInterrupt(uint8_t pin, std::function<void(void)> func, int mode)
+{
+    HRESULT hr;
+
+    hr = g_pins.verifyPinFunction(pin, FUNC_DIO, BoardPinsClass::NO_LOCK_CHANGE);
+
+    if (FAILED(hr))
+    {
+        ThrowError(hr, "Error occurred verifying pin: %d function: DIGITAL_IO, Error: %08x", pin, hr);

```

Arduino swallows all errors so it keeps running (although imperfectly). Is there a way we can log an error, without throwing (bringing down someone's system), or is this the preferred method for Windows?